bitcoin/src/blockdata/transaction.rs

```diff
@@ -1247,23 +1247,63 @@ pub fn predict_weight<I, O>(inputs: I, output_script_lens: O) -> Weight
                 let script_size = script_len + VarInt(script_len as u64).len();
                 (output_count + 1, total_scripts_size + script_size)
         });
-    let input_weight = partial_input_weight + input_count * 4 * (32 + 4 + 4);
-    let output_size = 8 * output_count + output_scripts_size;
-    let non_input_size =
-        // version:
-        4 +
-        // count varints:
-        VarInt(input_count as u64).len() +
-        VarInt(output_count as u64).len() +
-        output_size +
-        // lock_time
-        4;
-    let weight = if inputs_with_witnesses == 0 {
-        non_input_size * 4 + input_weight
-    } else {
-        non_input_size * 4 + input_weight + input_count - inputs_with_witnesses + 2
-    };
-    Weight::from_wu(weight as u64)
+    predict_weight_internal(input_count, partial_input_weight, inputs_with_witnesses, output_count, output_scripts_size)
+}
+
+crate::internal_macros::maybe_const_fn! {
+    fn predict_weight_internal(input_count: usize, partial_input_weight: usize, inputs_with_witnesses: usize, output_count: usize, output_scripts_size: usize) -> Weight {
+        let input_weight = partial_input_weight + input_count * 4 * (32 + 4 + 4);
+        let output_size = 8 * output_count + output_scripts_size;
+        let non_input_size =
+            // version:
+            4 +
+            // count varints:
+            VarInt(input_count as u64).len() +
+            VarInt(output_count as u64).len() +
+            output_size +
+            // lock_time
+            4;
+        let weight = if inputs_with_witnesses == 0 {
+            non_input_size * 4 + input_weight
+        } else {
+            non_input_size * 4 + input_weight + input_count - inputs_with_witnesses + 2
+        };
+        Weight::from_wu(weight as u64)
+    }
+}
+
+/// Predicts the weight of a to-be-constructed transaction in const context.
+///
+/// *Important: only available in Rust 1.46+*
+///
+/// This is a `const` version of [`predict_weight`] which only allows slices due to current Rust
+/// limitations around `const fn`. Because of these limitations it may be less efficient than
+/// `predict_weight` and thus is intended to be only used in `const` context.
+///
+/// Please see the documentation of `predict_weight` to learn more about this function.
+#[cfg(rust_v_1_46)]
+pub fn predict_weight_from_slices(inputs: &[InputWeightPrediction], output_script_lens: &[usize]) -> Weight {
```

Comment:
Docs say this is `const` but its `pub fn`, is `const` missing?

Reply:
:man_facepalming: that's what I get for trying to quickly push a change when tired.

Reply:
Lol, derp, good catch. (And it looks like it's fixed now.)